open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/impl/VRPConsentRequestHandler.java

```diff
@@ -0,0 +1,217 @@
+        }
+    }
+
+    /**
+     * Method to handle Variable Recurring Payment Consent Manage GET Request.
+     * This method retrieves and processes information related to a Variable Recurring Payment consent
+     * based on the provided consent ID. It validates the consent ID, checks if the consent exists,
+     * verifies if the consent belongs to the client making the request, and constructs a response payload
+     * containing relevant initiation retrieval details. The response status is set accordingly.
+     *
+     * @param consentManageData Object
+     *
+     */
+    @Override
+    public void handleConsentManageDelete(ConsentManageData consentManageData) {
+
+        ConsentManageUtil.handleConsentManageDelete(consentManageData);
+    }
+
+    /**
+     * Method to handle Variable Recurring Payment POST requests.
+     * This private method processes Variable Recurring Payment POST requests, creating a new consent
+     * based on the provided request payload. It performs the following actions:
+     *   - Creates a DetailedConsentResource representing the consent initiation.
+     *   - Stores consent attributes, including the idempotency key, for tracking and retrieval.
+     *   - Constructs the response payload containing initiation details and sets appropriate headers.
+     *   - Sets the response status to Create.
+     *
+     * @param consentManageData Object containing request details, including client ID, request payload, headers,
+     *                          and other relevant information.
+     */
+    private void handlePaymentPost(ConsentManageData consentManageData, Object request)
+            throws ConsentManagementException {
+
+        // Variable to store the created consent
+        DetailedConsentResource createdConsent;
+
+        JSONObject requestObject = (JSONObject) request;
+        // Create a ConsentResource representing the requested consent
+        ConsentResource requestedConsent = new ConsentResource(consentManageData.getClientId(),
+                requestObject.toJSONString(), ConsentExtensionConstants.VRP,
+                ConsentExtensionConstants.AWAITING_AUTH_STATUS);
+
+        // Create the consent and retrieve the detailed consent resource
+        createdConsent = ConsentExtensionsDataHolder.getInstance().getConsentCoreService()
+                .createAuthorizableConsent(requestedConsent, null,
+                        CREATED_STATUS, AUTH_TYPE_AUTHORIZATION, true);
+
+        //Set consent attributes for storing
+        Map<String, String> consentAttributes = new HashMap<>();
+        consentAttributes.put(ConsentExtensionConstants.IDEMPOTENCY_KEY, consentManageData.getHeaders()
+                .get(ConsentExtensionConstants.X_IDEMPOTENCY_KEY));
+        //Store consent attributes
+        ConsentServiceUtil.getConsentService().storeConsentAttributes(createdConsent.getConsentID(),
```

Comment:
We have stored the attributes here, But in latter part of the method we populate more values in to consentAttributes. These values might not be stored if we call `storeConsentAttributes` in the middle right? 

Ideally we should call `storeConsentAttributes()` , with all the attributes added to the `consentAttributes` variable.

Reply:
+1